language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/ExBudgeting.hs

```diff
@@ -103,8 +104,8 @@ data ExBudgetMode =
       Counting -- ^ For precalculation
     | Restricting ExRestrictingBudget -- ^ For execution, to avoid overruns
 
-class SpendBudget m where
-    spendBudget :: ExBudgetCategory -> WithMemory Term -> ExBudget -> m ()
+class SpendBudget m uni | m -> uni where
```

Comment:
Note: This works nicely because `m` contains `uni` as parameter.

Reply:
Thanks, I'll add that comment.